### PR TITLE
[fix](cloud) Fix constant folding for requests sent to BE without dis…

### PR DESCRIPTION
**Pull request by deardeng**
…tinguishing clusters

## Proposed changes

Issue Number: close #xxx

replace function getAllBackendIds by getBackendsByCurrentCluster

### CODE REVIEW

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnBE.java [hunk 1]
```diff
@@ -270,7 +271,8 @@ private static Map<String, Expression> evalOnBE(Map<String, Map<String, TExpr>>
             Map<String, Expression> constMap, ConnectContext context) {
         Map<String, Expression> resultMap = new HashMap<>();
         try {
-            List<Long> backendIds = Env.getCurrentSystemInfo().getAllBackendIds(true);
+            List<Long> backendIds = Env.getCurrentSystemInfo().getBackendsByCurrentCluster()
+                    .values().stream().filter(Backend::isAlive).map(Backend::getId).collect(Collectors.toList());
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnBE.java:275]: Please add a method getLiveBackendByCurrentCluster.

File: fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java [hunk 2]
```diff
@@ -363,7 +371,19 @@ public List<Long> getAllBackendIds() {
...
+            try {
+                beNumber = (int) getBackendsByCurrentCluster()
+                    .values().stream().filter(be -> {
+                        if (needAlive) {
+                            return be.isAlive();
+                        } else {
+                            // not filter, return all
+                            return true;
+                        }
+                    }).count();
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java:383]:
```suggestion
                        return needAlive ? be.isAlive() : true;
                    }).count();
```
```suggestion
                    .values().stream().filter(be -> {  return needAlive ? be.isAlive() : true; }).count();
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnBE.java (hunk 1 above)
Comment (gavinchou) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnBE.java:275]: getBackendsByCurrentCluster(boolean needAlive)

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**